lib/screens/onboarding/personal_details/onboarding_address_of_residence_screen.dart

```diff
@@ -82,6 +83,11 @@ class _OnboardingAddressOfResidenceScreenState extends State<OnboardingAddressOf
             );
           });
         }
+        
+        if (previousViewModel!.isAddressSaved == null && viewModel.isAddressSaved == true) {
+          Navigator.pushNamed(context, OnboardingMobileNumberScreen.routeName);
```

Comment:
I think that here we should have the `Navigator.pushNamedAndRemoveUntil` (or something like that), so when the user navigates to the mobile number screen doesn't have the option to go back